yt_dlp/postprocessor/xattrpp.py

```diff
@@ -29,21 +29,36 @@ class XAttrMetadataPP(PostProcessor):
         'user.dublincore.date': 'upload_date',
         'user.dublincore.contributor': 'uploader',
         'user.dublincore.format': 'format',
+        'com.apple.metadata:kMDItemWhereFroms': 'webpage_url',
         # We do this last because it may get us close to the xattr limits
         # (e.g., 4kB on ext4), and we don't want to have the other ones fail
         'user.dublincore.description': 'description',
         # 'user.xdg.comment': 'description',
     }
 
+    APPLE_PLIST_TEMPLATE = '''<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<array>
+\t<string>%s</string>
+</array>
+</plist>'''
+
+    def format_value(self, xattrname, infoname, value):
+        if infoname == 'upload_date':
+            return hyphenate_date(value)
+        if xattrname == 'com.apple.metadata:kMDItemWhereFroms':
+            return self.APPLE_PLIST_TEMPLATE % value
+        return value
+
     def run(self, info):
         mtime = os.stat(info['filepath']).st_mtime
         self.to_screen('Writing metadata to file\'s xattrs')
         for xattrname, infoname in self.XATTR_MAPPING.items():
             try:
                 value = info.get(infoname)
                 if value:
-                    if infoname == 'upload_date':
-                        value = hyphenate_date(value)
+                    value = self.format_value(xattrname, infoname, value)
```

Comment:
I would probably still inline the function call, since its just two `if`s but I think its fine like this as well